### PR TITLE
[CBR-461] Improve diagnostic for `NotEnoughMoney` error 

### PR DESCRIPTION
**Pull request by KtorZ**
## Description

When computing fees or making a transaction, it could happen that funds
are currently "locked" out because selected in a pending transaction. In
such case, the UTxO can't be spent until the transaction has been
accepted by (at least) a node. The API currently returns a
NotEnoughMoney error in such case, whereas the total balance (as
returned by /api/v1/wallets/{walletId} endpoint) might show enough
funds.

This refines a bit the `NotEnoughMoney` error to make the reason
of failure clearer without breaking (too much) the existing API.

NOTE: There's one "expected" failing integration test which actually shed the light on a newly discovered bug (0 fee transaction).

## Linked issue

[[CBR-461]](https://iohk.myjetbrains.com/youtrack/issue/CBR-461)

## Type of change

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🛠 New feature (non-breaking change which adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [ ] 🔨 New or improved tests for existing code
- [ ] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [x] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [x] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [x] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.
- [x] CHANGELOG entry has been added and is linked to the correct PR on GitHub.

## Testing checklist

- [x] I have added tests to cover my changes.
- ~~[ ] All new and existing tests passed.~~

### CODE REVIEW

**Comment by KtorZ** [comment]
Integration tests failing as "expected":

```
  integration/Util.hs:129:5: 
  1) Transactions fails if you can't cover fee with a transaction
       Falsifiable (after 1 test):
       
       predicate failed on: Right (Transaction {txId = AbstractHash 9459bb25a09439a0337857db0e2d3f95bf240a0a1608d60b9616072da4873ad6, txConfirmations = 0, txAmount = Coin {getCoin = 42}, txInputs = PaymentDistribution {pdAddress = Address {addrRoot = AbstractHash 42a18f1ee2cbbf37aad3370b27e43861c5330931c310157d75462dd7, addrAttributes = Attributes { data: AddrAttributes {aaPkDerivationPath = Just (HDAddressPayload {getHDAddressPayload = "\ENQ\SO{ \SI\RS\222\212\193\255\192c\166\200\v\204\182\240\201\161\213M\166V\134\160\172\212"}), aaStakeDistribution = BootstrapEraDistr} }, addrType = ATPubKey}, pdAmount = Coin {getCoin = 42}} :| [], txOutputs = PaymentDistribution {pdAddress = Address {addrRoot = AbstractHash f4cbe4cdeb6f727dd973b130c1233753415b69a601a3509c692eff45, addrAttributes = Attributes { data: AddrAttributes {aaPkDerivationPath = Just (HDAddressPayload {getHDAddressPayload = "\137\229\DC3(\US\229rmFE9\EOT+\129\GS\218w\166baK\156\213\EOT<\192-\147"}), aaStakeDistribution = BootstrapEraDistr} }, addrType = ATPubKey}, pdAmount = Coin {getCoin = 42}} :| [], txType = ForeignTransaction, txDirection = OutgoingTransaction, txCreationTime = 1538569020551488, txStatus = Applying})
```

Investigating in [[CBR-462]](https://iohk.myjetbrains.com/youtrack/issue/CBR-462)

File: CHANGELOG.md [hunk 1]
```diff
@@ -60,6 +60,7 @@
 
 - The codebase now relies on the package `cryptonite` (instead of `ed25519`) for Ed25519 implementation (CO-325)
 
+- **[API BREAKING CHANGE]** Improve diagnostic for `NotEnoughMoney` error (CBR-461)
```
Comment (adinapoli-iohk) [inline CHANGELOG.md]: Reminds me also of https://iohk.myjetbrains.com/youtrack/issue/CBR-449, heh.

File: wallet-new/src/Cardano/Wallet/API/V1/Swagger.hs [hunk 2]
```diff
@@ -350,7 +350,7 @@ $errors
   errors = T.intercalate "\n" rows
   rows =
     -- 'WalletError'
-    [ mkRow fmtErr $ NotEnoughMoney 1400
+    [ mkRow fmtErr $ NotEnoughMoney (ErrAvailableBalanceIsInsufficient 1400)
```
Comment (adinapoli-iohk) [inline wallet-new/src/Cardano/Wallet/API/V1/Swagger.hs]: Good lad!

File: CHANGELOG.md (hunk 1 above)
Comment (KtorZ) [reply CHANGELOG.md]: image

Whoops :trollface:

**Pull request merged**